### PR TITLE
Unk 020183 f0

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/berry_pots_app_internal.h [hunk 1]
```diff
@@ -83,7 +85,7 @@ typedef struct BerryPotsAppData {
     u16 totalBerryQuantity;
     u16 totalMulchQuantity;
     BgConfig *bgConfig;
-    void *unk8C;
+    TouchscreenListMenuSpawner *menuSpawner;
```
Comment (adrienntindall) [inline include/berry_pots_app_internal.h:88]: is spawner really the best term here?

File: include/mail_misc.h [hunk 2]
```diff
@@ -66,7 +67,7 @@ void sub_02090D8C(EasyChatArgs *args, MailMessage *msg1, MailMessage *msg2);
 BOOL sub_02090DC0(EasyChatArgs *args, MailMessage *msg1, MailMessage *msg2);
 void sub_02090E04(EasyChatArgs *args, MailMessage *msg1, MailMessage *msg2);
 int sub_02090E44(EasyChatArgs *args);
-void sub_02090E5C(EasyChatArgs *args, u32 a1);
+void sub_02090E5C(EasyChatArgs *args, MenuInputState a1);
```
Comment (adrienntindall) [inline include/mail_misc.h:70]: a1 -> menuInputState

File: include/menu_input_state.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#define POKEHEARTGOLD_MENU_INPUT_STATE_H
+
+#include "global.h"
+
+typedef enum MenuInputState {
+    MENU_INPUT_STATE_BUTTONS,
+    MENU_INPUT_STATE_TOUCH,
+} MenuInputState;
+
+typedef struct MenuInputStateMgr {
```
Comment (adrienntindall) [inline include/menu_input_state.h:11]: Why is this a struct and can you get it to match without it?

File: include/options_app.h [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
 
+#include "menu_input_state.h"
 #include "options.h"
 #include "overlay_manager.h"
 
 typedef struct OptionsApp_Args {
     u32 unk0;
     Options *options;
-    BOOL *unk8;
+    MenuInputStateMgr *unk8;
```
Comment (adrienntindall) [inline include/options_app.h:11]: unk8 -> menuInputStateMgr
Also if Mgr = Manager we should write it as Man (shorthand) or straight up in long hand

File: include/overlay_55.h [hunk 5]
```diff
@@ -11,7 +11,7 @@ typedef struct UnkStruct_ov55_021E5B08 {
     u16 unk0;
     u8 mailMessageIdx;
     Options *options;
-    u32 *unk8;
+    MenuInputStateMgr *unk8;
```
Comment (adrienntindall) [inline include/overlay_55.h:14]: name unk8

File: include/party_menu.h [hunk 6]
```diff
@@ -484,7 +484,7 @@ u32 sub_0207CAA8(void);
 void sub_0207CAAC(HeapID heapId, u16 *a1, u16 *a2, u16 *a3);
 void PartyMenu_FormChangeScene_End(PartyMenu *partyMenu);
 void PartyMenu_DeleteContextMenuAndList(PartyMenu *partyMenu);
-void sub_0207CB3C(PartyMenu *partyMenu, BOOL a1);
+void sub_0207CB3C(PartyMenu *partyMenu, MenuInputState a1);
```
Comment (adrienntindall) [inline include/party_menu.h:487]: a1 -> menuInputState

File: src/alph_puzzle.c [hunk 7]
```diff
@@ -660,15 +660,15 @@ static void AlphPuzzle_ScreenOff(void) {
 }
 
 static void AlphPuzzle_InitTextOptionsAndPuzzleIndex(AlphPuzzleData *data) {
-    data->menuIgnoreTouchFlag = sub_020183F0(data->args->menuInputStatePtr);
+    data->menuIgnoreTouchFlag = MenuInputStateMgr_GetState(data->args->menuInputStatePtr);
```
Comment (adrienntindall) [inline src/alph_puzzle.c:663]: TouchFlag -> TouchState?

File: src/application/view_photo.c [hunk 8]
```diff
@@ -198,10 +198,10 @@ static ViewPhotoInputResponse ViewPhotoSysTask_HandleInput(ViewPhotoSysTaskData
     if (response == VIEW_PHOTO_INPUT_NOTHING) {
         response = ViewPhotoSysTask_GetKeyInput(viewPhoto);
         if (response == VIEW_PHOTO_INPUT_NOTHING) {
-            viewPhoto->lastInputWasTouch = FALSE;
+            viewPhoto->lastInputWasTouch = MENU_INPUT_STATE_BUTTONS;
```
Comment (adrienntindall) [inline src/application/view_photo.c:201]: lastInputWasTouch -> inputState

File: src/options_app.c [hunk 9]
```diff
@@ -340,11 +340,11 @@ BOOL OptionsMenu_Init(OVY_MANAGER *manager, int *state) {
     data->options.buttonMode  = Options_GetButtonMode(args->options);
     data->options.frame       = Options_GetFrame(args->options);
 
-    data->unk20               = args->unk8;
+    data->menuInputPtr        = args->unk8;
```
Comment (adrienntindall) [inline src/options_app.c:343]: unk8 -> menuInputPtr

File: src/options_app.c [hunk 10]
```diff
@@ -340,11 +340,11 @@ BOOL OptionsMenu_Init(OVY_MANAGER *manager, int *state) {
...
     data->playerOptionsUnused = args->options;
     data->heapId              = HEAP_ID_OPTIONS_APP;
     data->playerOptions       = args->options;
-    data->unk320              = sub_020183F0(data->unk20);
+    data->unk320              = MenuInputStateMgr_GetState(data->menuInputPtr);
```
Comment (adrienntindall) [inline src/options_app.c:347]: unk320 -> menuInputState

File: src/party_menu.c [hunk 11]
```diff
@@ -91,7 +91,7 @@ static BOOL PartyMenu_HandleDpadInput(PartyMenu *partyMenu);
 static u8 PartyMenu_GetSelectionInDirection(PartyMenu *partyMenu, u8 *px, u8 *py, u8 direction);
 static u8 PartyMenu_GetNewSelectionFromTable(PartyMenu *partyMenu, u8 *px, u8 *py, const u8 *table);
 static void sub_0207AC20(PartyMenu *partyMenu);
-static int sub_0207AC70(PartyMenu *partyMenu, BOOL a1);
+static int sub_0207AC70(PartyMenu *partyMenu, MenuInputState a1);
```
Comment (adrienntindall) [inline src/party_menu.c:94]: name a1

File: src/party_menu.c [hunk 12]
```diff
@@ -1443,7 +1443,7 @@ static void sub_0207AC20(PartyMenu *partyMenu) {
     }
 }
 
-static int sub_0207AC70(PartyMenu *partyMenu, BOOL a1) {
+static int sub_0207AC70(PartyMenu *partyMenu, MenuInputState a1) {
```
Comment (adrienntindall) [inline src/party_menu.c:1446]: name a1

File: src/party_menu.c [hunk 13]
```diff
@@ -2712,9 +2712,9 @@ void PartyMenu_DeleteContextMenuAndList(PartyMenu *partyMenu) {
     ListMenuItems_Delete(partyMenu->listMenuItems);
 }
 
-void sub_0207CB3C(PartyMenu *partyMenu, BOOL a1) {
+void sub_0207CB3C(PartyMenu *partyMenu, MenuInputState a1) {
```
Comment (adrienntindall) [inline src/party_menu.c:2715]: name a1

File: src/touch_save_app.c [hunk 14]
```diff
@@ -497,8 +497,8 @@ static BOOL TouchSaveApp_ShouldPrintAlternateSavingMessage(TouchSaveAppData *dat
     return Save_NumModifiedPCBoxesIsMany(data->fieldSystem->saveData);
 }
 
-static void ov30_0225DC00(BOOL *a0, u32 a1) {
-    sub_02018410(a0, a1);
+static void ov30_0225DC00(MenuInputStateMgr *stateMgr, MenuInputState state) {
```
Comment (adrienntindall) [inline src/touch_save_app.c:500]: can prolly name this as just a static version of set state

File: src/touchscreen_list_menu.c [hunk 15]
```diff
@@ -0,0 +1,429 @@
...
+static void TouchscreenListMenu_PrintOptions(TouchscreenListMenu *menu);
+static void TouchscreenListMenu_ToggleButtonPalette(TouchscreenListMenu *menu, u8 index, BOOL selected);
+static void TouchscreenListMenu_EraseTilemap(TouchscreenListMenu *menu);
+static void TouchscreenListMenu_DrawButtonsAndTransfer(TouchscreenListMenu *menu, u8 cursorPos);
+static int TouchscreenListMenu_HandleTouchInput(TouchscreenListMenu *menu, BOOL *flagRet);
+static int TouchscreenListMenu_HandleKeyInput(TouchscreenListMenu *menu);
+static void TouchscreenListMenu_PlaySE(TouchscreenListMenu *menu, u16 sndseq);
+static void TouchscreenListMenu_InvokeCallback(TouchscreenListMenu *menu, int a1);
+
+TouchscreenListMenuSpawner *TouchscreenListMenuSpawner_Create(HeapID heapId, PaletteData *paletteData) {
```
Comment (adrienntindall) [inline src/touchscreen_list_menu.c:32]: Ideally just rename spawner to manager and remove the current manager as a struct

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: include/berry_pots_app_internal.h (hunk 1 above)
Comment (PikalaxALT) [reply include/berry_pots_app_internal.h:88]: i'm open to suggestions

File: include/menu_input_state.h (hunk 3 above)
Comment (adrienntindall) [reply include/menu_input_state.h:11]: considering this is gf, yes

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**